tests/phpunit/tests/blocks/wpGetBlockPatterns.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * Tests for _wp_get_block_patterns.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ * @since 6.4.0
+ *
+ * @group blocks
+ *
+ * @covers ::_wp_get_block_patterns
+ */
+class Tests_Blocks_wpGetBlockPatterns extends WP_UnitTestCase {
+	/**
+	 * @ticket 59490
+	 *
+	 * @dataProvider data_wp_get_block_patterns
+	 */
+	public function test_wp_get_block_patterns( $theme, $expected ) {
+		$patterns = _wp_get_block_patterns( wp_get_theme( $theme ) );
+		$this->assertSameSets( $expected, $patterns );
+	}
+
+	/**
+	 * @ticket 59490
+	 */
+	public function test_delete_theme_cache() {
+		$theme = wp_get_theme( 'block-theme-patterns' );
+		_wp_get_block_patterns( $theme );
+		$transient = get_transient( 'wp_theme_patterns_block-theme-patterns' );
+		$this->assertSameSets(
+			array(
+				'version'  => '1.0.0',
+				'patterns' => array(
+					'cta.php' => array(
+						'title'       => 'Centered Call To Action',
+						'slug'        => 'block-theme-patterns/cta',
+						'description' => '',
+						'categories'  => array( 'call-to-action' ),
+					),
+				),
+			),
+			$transient,
+			'The transient for block theme patterns should be set'
+		);
+		$theme->cache_delete();
+		$transient = get_transient( 'wp_theme_patterns_block-theme-patterns' );
+		$this->assertFalse(
+			$transient,
+			'The transient for block theme patterns should have been cleared'
+		);
```

Comment:
```suggestion
		$this->assertSameSets(
			array(
				'version'  => '1.0.0',
				'patterns' => array(
					'cta.php' => array(
						'title'       => 'Centered Call To Action',
						'slug'        => 'block-theme-patterns/cta',
						'description' => '',
						'categories'  => array( 'call-to-action' ),
					),
				),
			),
			get_transient( 'wp_theme_patterns_block-theme-patterns' ),
			'The transient for block theme patterns should be set'
		);

		$theme->cache_delete();

		$this->assertFalse(
			get_transient( 'wp_theme_patterns_block-theme-patterns' ),
			'The transient for block theme patterns should have been cleared'
		);
```
No need to store in a variable, it's still clear what it is, and the line length is comparable to nearby lines anyway. We can add space around the `$theme->delete()` to make it clear when the change should happen.